clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -276,6 +276,7 @@ public class ProducerConfig extends AbstractConfig {
     public static final String RETRIES_CONFIG = CommonClientConfigs.RETRIES_CONFIG;
     private static final String RETRIES_DOC = "Setting a value greater than zero will cause the client to resend any record whose send fails with a potentially transient error."
             + " Note that this retry is no different than if the client resent the record upon receiving the error."
+            + " Setting a value of zero will make that transient errors won't be retried, and will be propagated to the application to be handled."
```

Comment:
Sounds redundant to the next sentence? `Produce requests will be failed before the number of retries has been exhausted` -- with `retries=0` is implies that transient error will be throw into the application right away, right?

I agree that the new sentence is easier to read, but should we do a larger rewrite to avoid redundancy?